### PR TITLE
Feature/js clicks to functions

### PR DESCRIPTION
**Pull request by Peleccotur**
Updated clicks and websocket to trigger functions to prepare for sending variables number and timers

### CODE REVIEW

**Comment by Peleccotur** [comment]
OH, and I also added a try/catch on the JSON.Parse. It was throwing errors on the clock updates

File: public/js/logic.js [hunk 1]
```diff
@@ -6,35 +6,61 @@ $(document).ready(function(){
...
+    clearAllChests();
   });
 
-  // Update count when adding, removing, clearing
-  $("#controls button").click(function() {
-    $("#count").val(chestCount);
+  // Remove the chest you click on
+  $("#container").on("click", "div.chest-wrapper", function(){
+    // "on" delegated element events need a static callback function
+    $(this).hide("fast", done());
```
Comment (amaisano) [inline public/js/logic.js]: We lost the `event` arg in done()

File: public/js/logic.js [hunk 2]
```diff
@@ -6,35 +6,61 @@ $(document).ready(function(){
...
+    chestCount = 0;
+    updateCount();
+  }
+
+  function done(event) {
+    // Event is actually propagated on the .chest element
+    $(event.target).parents('div.chest-wrapper').remove();
+    chestCount--;
+    localStorage.setItem('count', chestCount);
+    $("#count").val(chestCount);
```
Comment (amaisano) [inline public/js/logic.js]: These should be refactored into update Count right?

**Review by amaisano — CHANGES_REQUESTED** [review]

File: public/js/logic.js (hunk 1 above)
Comment (Peleccotur) [reply public/js/logic.js]: Not lost, I removed it. Its deprecated to specify since its implied.

File: public/js/logic.js (hunk 2 above)
Comment (Peleccotur) [reply public/js/logic.js]: ah, missed that. Indeed

File: public/js/logic.js (hunk 1 above)
Comment (amaisano) [reply public/js/logic.js]: Ah. My IDE gave me deprecation warnings about that and my 3am mind didn't figure it out.

File: public/js/logic.js [hunk 3]
```diff
@@ -15,42 +15,58 @@ $(document).ready(function(){
...
   });
 
   // Remove last chest
   $("#remove").click(function(){
-    if (chestCount > 0) {
-      container.children().last().remove();
-      chestCount--;
-      localStorage.setItem('count', chestCount);
-    }
+    deleteChests(1);
```
Comment (amaisano) [inline public/js/logic.js]: Could you make this function "removeChests" so that the language matches the payload options and the UI?

**Review by amaisano — CHANGES_REQUESTED** [review]
Just one naming item. Otherwise RTBM.

**Review by amaisano — APPROVED** [review]

**Pull request merged**